facades/WorkspaceValidation/src/test/java/org/terasology/workspace/validation/WorkspaceBehaviorsTests.java

```diff
@@ -0,0 +1,142 @@
+                            Stream.of(engine).map((e) ->
+                                    DynamicTest.dynamicTest("setup", () -> {
+                                        e.setup();
+                                        assetManagerRef.set(e.getHostContext().get(AssetManager.class));
+                                    })
+                            ),
+                            Stream.of(DynamicContainer.dynamicContainer("Assets tests", Stream.of(
+                                    asset(
+                                            assetManagerRef,
+                                            BehaviorTree.class,
+                                            (b) -> Assertions.assertNotNull(b.getData())
+                                    ),
+                                    asset(
+                                            assetManagerRef,
+                                            Prefab.class,
+                                            prefab -> prefab.hasComponent(BehaviorComponent.class),
+                                            prefab -> Assertions.assertNotNull(prefab.getComponent(BehaviorComponent.class).tree)
+
+                                    )))),
+                            Stream.of(DynamicTest.dynamicTest("tearDown", engine::tearDown))));
+                });
+    }
+
+    private <T extends AssetData, A extends Asset<T>> DynamicContainer asset(AtomicReference<AssetManager> assetManager,
+                                                                             Class<A> assetClazz,
+                                                                             Consumer<A> validator) {
+        return asset(assetManager, assetClazz, (a) -> true, validator);
+
+    }
+
+    private <T extends AssetData, A extends Asset<T>> DynamicContainer asset(AtomicReference<AssetManager> assetManager,
+                                                                             Class<A> assetClazz,
+                                                                             Predicate<A> filter,
+                                                                             Consumer<A> validator) {
+
+        return DynamicContainer.dynamicContainer(String.format("AssetType: %s", assetClazz.getSimpleName()),
+                DynamicTest.stream(
+                        Stream.generate(() -> assetManager.get().getAvailableAssets(assetClazz).stream())
+                                .limit(1)
+                                .flatMap(s -> s)
+                                .filter(urn -> filter.test(assetManager.get().getAsset(urn, assetClazz).get())),
+                        ResourceUrn::toString,
+                        urn -> {
+                            A asset = (A) assetManager.get().getAsset(urn, assetClazz).get();
+                            validator.accept(asset);
+                        }
+                ));
+
+    }
+
+    public static class EnginesAccessor extends Engines {
```

Comment:
What's going on here? Why do we have a subclass that doesn't actually have any custom behavior? 😰

Reply:
It is special temporary magic for accessing protected methods. ;)
Engines is nice enough, but i cannot access to EngineCleaner